### PR TITLE
Replace deprecated isAllowed(...) with userHasRight(...)

### PR DESCRIPTION
**Pull request by paladox**
Bug: T5008

### CODE REVIEW

File: includes/ManageWiki.php [hunk 1]
```diff
@@ -27,11 +27,12 @@ public static function listModules() {
 	public static function checkPermission( RemoteWiki $rm, User $user, string $perm = "" ) {
 		$maxPerm = ( (bool)$perm ) ? $perm : 'managewiki';
 
-		if ( $rm->isLocked() && !$user->isAllowed( 'managewiki-restricted' ) ) {
+		$mwService = MediaWiki\MediaWikiServices::getInstance()->getPermissionManager();
```
Comment (Southparkfan) [inline includes/ManageWiki.php:30]: $mwService -> $permissionManager

File: includes/ManageWikiHooks.php [hunk 2]
```diff
@@ -353,7 +353,8 @@ public static function fnNewSidebarItem( $skin, &$bar ) {
 		$append = '';
 
 		$user = $skin->getUser();
-		if ( !$user->isAllowed( 'managewiki' ) ) {
+		$mwService = MediaWiki\MediaWikiServices::getInstance()->getPermissionManager();
```
Comment (Southparkfan) [inline includes/ManageWikiHooks.php:356]: $mwService -> $permissionManager

File: includes/formFactory/ManageWikiFormFactoryBuilder.php [hunk 3]
```diff
@@ -94,6 +94,8 @@ private static function buildDescriptorCore(
 			]
 		];
 
+		$mwService = MediaWiki\MediaWikiServices::getInstance()->getPermissionManager();
```
Comment (Southparkfan) [inline includes/formFactory/ManageWikiFormFactoryBuilder.php:97]: $mwService -> $permissionManager

File: includes/formFactory/ManageWikiFormFactoryBuilder.php [hunk 4]
```diff
@@ -955,7 +958,8 @@ private static function submissionCore(
 				$changedArray[] = ( $newInactive ) ? 'inactive' : 'active';
 			}
 
-			if ( $newInactiveExempt && $context->getUser()->isAllowed( 'managewiki-restricted' ) ) {
+			$mwService = MediaWiki\MediaWikiServices::getInstance()->getPermissionManager();
```
Comment (Southparkfan) [inline includes/formFactory/ManageWikiFormFactoryBuilder.php:961]: $mwService -> $permissionManager

File: includes/formFactory/ManageWikiFormFactoryBuilder.php [hunk 5]
```diff
@@ -283,7 +285,8 @@ private static function buildDescriptorSettings(
 						break;
 				}
 
-				$disabled = !( !$set['restricted'] || ( $set['restricted'] && $context->getUser()->isAllowed( 'managewiki-restricted' ) ) );
+				$mwService = MediaWiki\MediaWikiServices::getInstance()->getPermissionManager();
```
Comment (Southparkfan) [inline includes/formFactory/ManageWikiFormFactoryBuilder.php:288]: $mwService -> $permissionManager

File: includes/formFactory/ManageWikiFormFactoryBuilder.php [hunk 6]
```diff
@@ -1066,7 +1070,8 @@ private static function submissionSettings(
 
 		foreach ( $wgManageWikiSettings as $name => $set ) {
 			$current = $wiki->getSettingsValue( $name );
-			$mwAllowed = ( $set['restricted'] && $context->getUser()->isAllowed( 'managewiki-restricted' ) || !$set['restricted'] );
+			$mwService = MediaWiki\MediaWikiServices::getInstance()->getPermissionManager();
```
Comment (Southparkfan) [inline includes/formFactory/ManageWikiFormFactoryBuilder.php:1073]: $mwService -> $permissionManager

File: includes/helpers/ManageWikiRequirements.php [hunk 7]
```diff
@@ -24,7 +24,8 @@ public static function process( string $dbname, array $actions, IContextSource $
 
 	private static function permissions( array $data, IContextSource $context ) {
 		foreach ( $data as $perm ) {
-			if ( !$context->getUser()->isAllowed( $perm ) ) {
+			$mwService = MediaWiki\MediaWikiServices::getInstance()->getPermissionManager();
```
Comment (Southparkfan) [inline includes/helpers/ManageWikiRequirements.php:27]: $mwService -> $permissionManager

File: includes/specials/SpecialManageWiki.php [hunk 8]
```diff
@@ -137,7 +137,8 @@ private function reusableFormDescriptor( string $module, array $options ) {
 		$selectForm = HTMLForm::factory( 'ooui', $hidden + $selector, $this->getContext(), 'selector' );
 		$selectForm->setMethod( 'post' )->setFormIdentifier( 'selector' )->setSubmitCallback( [ $this, 'reusableFormSubmission' ] )->prepareForm()->show();
 
-		if ( $this->getContext()->getUser()->isAllowed( 'managewiki' ) ) {
+		$mwService = MediaWiki\MediaWikiServices::getInstance()->getPermissionManager();
```
Comment (Southparkfan) [inline includes/specials/SpecialManageWiki.php:140]: $mwService -> $permissionManager

File: includes/specials/SpecialManageWikiDefaultPermissions.php [hunk 9]
```diff
@@ -42,7 +42,8 @@ public function buildMainView() {
 		$selectForm = HTMLForm::factory( 'ooui', $groupSelector, $this->getContext(), 'groupSelector' );
 		$selectForm->setMethod('post' )->setFormIdentifier( 'groupSelector' )->setSubmitCallback( [ $this, 'onSubmitRedirectToPermissionsPage' ] )->prepareForm()->show();
 
-		if ( $this->getContext()->getUser()->isAllowed( 'managewiki-editdefault' ) ) {
+		$mwService = MediaWiki\MediaWikiServices::getInstance()->getPermissionManager();
```
Comment (Southparkfan) [inline includes/specials/SpecialManageWikiDefaultPermissions.php:45]: $mwService -> $permissionManager

**Comment by JohnFLewis** [comment]
Can non-static classes have it defined once only as well please? No need to duplicate calls when not needed

File: includes/specials/SpecialManageWikiDefaultPermissions.php (hunk 9 above)
Comment (paladox) [reply includes/specials/SpecialManageWikiDefaultPermissions.php]: Thanks, done.

File: includes/specials/SpecialManageWiki.php (hunk 8 above)
Comment (paladox) [reply includes/specials/SpecialManageWiki.php]: Thanks, done.

File: includes/helpers/ManageWikiRequirements.php (hunk 7 above)
Comment (paladox) [reply includes/helpers/ManageWikiRequirements.php]: Thanks, done.

File: includes/formFactory/ManageWikiFormFactoryBuilder.php (hunk 6 above)
Comment (paladox) [reply includes/formFactory/ManageWikiFormFactoryBuilder.php]: Thanks, done.

File: includes/formFactory/ManageWikiFormFactoryBuilder.php (hunk 5 above)
Comment (paladox) [reply includes/formFactory/ManageWikiFormFactoryBuilder.php]: Thanks, done.

File: includes/formFactory/ManageWikiFormFactoryBuilder.php (hunk 4 above)
Comment (paladox) [reply includes/formFactory/ManageWikiFormFactoryBuilder.php]: Thanks, done.

File: includes/formFactory/ManageWikiFormFactoryBuilder.php (hunk 3 above)
Comment (paladox) [reply includes/formFactory/ManageWikiFormFactoryBuilder.php]: Thanks, done.

File: includes/ManageWikiHooks.php (hunk 2 above)
Comment (paladox) [reply includes/ManageWikiHooks.php]: Thanks, done.

File: includes/ManageWiki.php (hunk 1 above)
Comment (paladox) [reply includes/ManageWiki.php]: Thanks, done.

**Pull request merged**